ipc/provider/src/lib.rs

```diff
@@ -290,6 +290,16 @@ impl IpcProvider {
             .await
     }
 
+    pub async fn bootstrapped(&mut self, subnet: &SubnetID) -> anyhow::Result<bool> {
+        let parent = subnet.parent().ok_or_else(|| anyhow!("no parent found"))?;
+        let conn = match self.connection(&parent) {
+            None => return Err(anyhow!("target parent subnet not found")),
+            Some(conn) => conn,
+        };
```

Comment:
I see this error handling appear ~11 times in this file. Can it be factored out?